### PR TITLE
[runtime] Fail gracefully when method has conflicting attributes

### PR DESCRIPTION
**Pull request by alexanderkyte**

### CODE REVIEW

**Comment by lambdageek** [comment]
Is `g_error()` the best way we have for signaling AOT failures?

**Comment by lambdageek** [comment]
Also, can't this happen with normal JITing without AOT?

**Comment by alexanderkyte** [comment]
So it seems to not really be something the JIT hits, it just runs correctly when both attributes are mixed. We have so few callers that I can push this off to them though for better reporting, I'll do that. All of the callers but one just pass it to g_error themselves, but that's fine.

**Comment by kumpera** [comment]
The long form fix would be to MonoError'ify mono_marshal_get_managed_wrapper.

File: mono/metadata/marshal.c [hunk 1]
```diff
@@ -8659,7 +8661,7 @@ mono_marshal_set_callconv_from_modopt (MonoMethod *method, MonoMethodSignature *
  * If target_handle==0, the wrapper info will be a WrapperInfo structure.
  */
 MonoMethod *
-mono_marshal_get_managed_wrapper (MonoMethod *method, MonoClass *delegate_klass, uint32_t target_handle)
+mono_marshal_get_managed_wrapper (MonoMethod *method, MonoClass *delegate_klass, uint32_t target_handle, MonoError *external_error)
 {
```
Comment (vargaz) [inline mono/metadata/marshal.c]: This should just be error, and the local error should be removed.

File: mono/mini/aot-runtime.c [hunk 2]
```diff
@@ -1214,7 +1214,7 @@ decode_method_ref_with_target (MonoAotModule *module, MethodRef *ref, MonoMethod
 			klass = decode_klass_ref (module, p, &p, error);
 			if (!klass)
 				return FALSE;
-			ref->method = mono_marshal_get_managed_wrapper (m, klass, 0);
+			ref->method = mono_marshal_get_managed_wrapper (m, klass, 0, error);
 			break;
```
Comment (vargaz) [inline mono/mini/aot-runtime.c]: This should check error.

**Pull request merged**